File: P7-NotesAppAPI/NotesApp.Domain/Note.cs

namespace NotesApp.Domain
{
    public class Note
    {
        public int Id { get; set; }
        public string Name { get; set; } = null!;
        public string IdentityUserId { get; set; } = null!;
        public int CategoryId { get; set; }
        public Category Category { get; set; } = null!;
        public List<ToDoTask>? ToDoTasks { get; set; }
    }
}

File: P8-NotesAppPage/src/app/_interfaces/Note/Note.ts
import { Category } from "../Category/Category";
import { ToDoTask } from "../ToDoTask/ToDoTask";


export interface Note {
    id: number;
    name: string;
    identityUserId: string;
    categoryId: number;
    category: Category;
    toDoTasks: ToDoTask[] | null;
}